SingularityService/src/main/java/com/hubspot/singularity/resources/RequestResource.java

```diff
@@ -201,8 +202,17 @@ public SingularityRequestParent bounce(@ApiParam("The request ID to bounce") @Pa
 
     requestManager.bounce(requestWithState.getRequest(), System.currentTimeMillis(), JavaUtils.getUserEmail(user), message);
 
+    SingularityBounceRequest defaultBounceRequest = bounceRequest.or(SingularityBounceRequest.defaultRequest());
+    if (!defaultBounceRequest.getDurationMillis().isPresent()) {
+      final Long durationMillis = TimeUnit.MINUTES.toMillis(configuration.getDefaultBounceExpirationMinutes());
```

Comment:
When using it as a type in the Optional<> you have to use the generic object (`Long`) in that case. A primitive does not extend any `Object` so it cannot be the type argument there. But, when it isn't required to have the generic object (like in your case, just setting a variable and using the value) it is preferred to use the primitive. You can pass that primitive (`long`) as an arg for something that takes `Long` just fine. (i.e. `Optional.of(yourVar)`)